zipkin-autoconfigure/storage-influxdb/src/main/java/zipkin/autoconfigure/storage/influxdb/ZipkinInfluxDBStorageAutoConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.autoconfigure.storage.influxdb;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import zipkin.internal.V2StorageComponent;
+import zipkin.storage.StorageComponent;
+import zipkin2.storage.influxdb.InfluxDBStorage;
+
+@Configuration
+@EnableConfigurationProperties(ZipkinInfluxDBStorageProperties.class)
+@ConditionalOnProperty(name = "zipkin.storage.type", havingValue = "influxdb-http")
+@ConditionalOnMissingBean(StorageComponent.class)
+public class ZipkinInfluxDBStorageAutoConfiguration {
+
+  @Bean
+  @ConditionalOnMissingBean
+  V2StorageComponent storage(
+    ZipkinInfluxDBStorageProperties properties,
+    @Value("${zipkin.storage.strict-trace-id:true}") boolean strictTraceId) {
+    InfluxDBStorage storage = properties.toBuilder().strictTraceId(strictTraceId).build();
+    return V2StorageComponent.create(storage);
+  }
```

Comment:
add another bean like below as some components specifically look for v2 variants. V2StorageComponent is an adapter that is actually an instance of the v1 api.

```java
  @Bean InfluxDBStorage v2Storage(V2StorageComponent component) {
    return (InfluxDBStorage) component.delegate();
  }
```

Reply:
done